### PR TITLE
Add ConstOverflowingMul, ConstWrappingMul, ConstCheckedMul, ConstSatu…

### PR DESCRIPTION
**Pull request by kaidokert**
…ratingMul

#58 

- Define const mul traits in const_numtrait.rs
- Implement for primitive types (u8, u16, u32, u64, u128)
- Implement for FixedUInt with const_array_mul
- num_traits wrappers delegate to const versions
- Add tests including nightly const evaluation

## Summary by Sourcery

Add const-capable multiplication traits and implementations for primitive integers and FixedUInt, and update wrappers and tests to exercise const multiplication behavior.

New Features:
- Introduce ConstOverflowingMul, ConstWrappingMul, ConstCheckedMul, and ConstSaturatingMul traits for const-evaluated multiplication semantics.
- Implement const multiplication traits for u8, u16, u32, u64, u128, and FixedUInt using const_array_mul.
- Expose const helper functions to invoke the const multiplication traits on FixedUInt for use in tests.

Enhancements:
- Update num_traits multiplication wrappers for FixedUInt to delegate to the new const multiplication trait implementations.

Tests:
- Add runtime and nightly const-evaluation tests covering const wrapping, checked, saturating, and overflowing multiplication for FixedUInt.

## Summary by CodeRabbit

* **New Features**
  * Added compile-time multiplication support with overflowing, wrapping, checked, and saturating behaviors; preserves existing runtime multiplication behavior.

* **Tests**
  * Added tests covering const-evaluated multiplication, overflow cases, and related edge conditions.

### CODE REVIEW

File: src/fixeduint/mul_div_impl.rs [hunk 1]
```diff
@@ -1,7 +1,9 @@
-use num_traits::ops::overflowing::OverflowingMul;
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint/mul_div_impl.rs]: @gemini-code-assist code compiles fine - see CI compile+test results
